### ISSUE TITLE
java.lang.String cannot be cast to net.sf.json.JSONObject

### ISSUE THREAD

**Issue by mr-const** [jenkinsci/saltstack-plugin#8]
I'm trying to run simple runner command, but it constantly fails. Unfortunately I can't debug it myself, because not enough debug information provided. When I run same task from command-line curl - it works ok.

Here is a curl log:
```
curl -sSk https://mysalt.domain.tld/salt/ -H "X-Auth-Token: 123456789ABCDEF...." -d "client=runner" -d "fun=fileserver.update"
{"return": [true]}
```

Here is a jenkins' failure log:
```
Building remotely on dsg-buildman-mac-00 (android ios mac) in workspace /Users/jenkins/workspace/RS-Server-Deploy
Sending JSON: [{"client":"runner","mods":"","tgt":"notused","expr_form":"glob","fun":"cmd.run","arg":["fileserver.update"]}]
ERROR: Build step failed with exception
java.lang.ClassCastException: java.lang.String cannot be cast to net.sf.json.JSONObject
	at com.waytta.Utils.validateFunctionCall(Utils.java:141)
	at com.waytta.SaltAPIBuilder.perform(SaltAPIBuilder.java:339)
	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:782)
	at hudson.model.Build$BuildExecution.build(Build.java:205)
	at hudson.model.Build$BuildExecution.doRun(Build.java:162)
	at hudson.model.AbstractBuild$AbstractBuildExecution.run(AbstractBuild.java:534)
	at hudson.model.Run.execute(Run.java:1738)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
	at hudson.model.ResourceController.execute(ResourceController.java:98)
	at hudson.model.Executor.run(Executor.java:410)
Build step 'Send a message to Salt API' marked build as failure
Finished: FAILURE
```

**Comment by mr-const** [comment]
I've run curl with JSON parameters, which sends saltstack plugin and finally could read error description. It seems like my mistake in a project configuration. However better debug diagnostics of the plugin would be very helpful :)

```
curl -X POST https://mysalt.domain.tld/salt/ -H "X-Auth-Token: 12345" -H "Content-Type: application/json" -d "[{\"client\":\"runner\",\"mods\":\"\",\"tgt\":\"notused\",\"expr_form\":\"glob\",\"fun\":\"cmd.run\",\"arg\":[\"fileserver.update\"]}]" 
{"return": ["Exception occurred in runner cmd.run: Traceback (most recent call last):\n  File \"/usr/lib/python2.7/dist-packages/salt/client/mixins.py\", line 267, in low\n    verify_fun(self.functions, fun)\n  File \"/usr/lib/python2.7/dist-packages/salt/utils/lazy.py\", line 20, in verify_fun\n    raise salt.exceptions.CommandExecutionError(lazy_obj.missing_fun_string(fun))\nCommandExecutionError: Module 'cmd' is not available.\n"]}
```

**Comment by mchugh19** [comment]
You are correct, we were not handling string returns.  As mentioned in the other bug report, I've whipped up a quick fix. Can you try out: https://github.com/jenkinsci/saltstack-plugin/releases/tag/v1.6.1-beta ? It will hopefully correct this issue. 

Thanks much!

**Comment by mr-const** [comment]
It works, now it's possible to see salt's answer in a jenkins' console. 
Thank you.

**Comment by mchugh19** [comment]
Woo hoo!

In the just released 1.6.1 build. Should hit the jenkins mirror and be available in the next 24 hours.
